cmd/nodegenesis.go

```diff
@@ -4,44 +4,53 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+	"path/filepath"
 
 	"github.com/0xPolygon/cdk-contracts-tooling/config"
+	"github.com/0xPolygon/cdk-contracts-tooling/rollupmanager"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/urfave/cli/v2"
 )
 
+const (
+	genesisAllocsFileName = "allocs.json"
+	fepDirName            = "fep"
+	ppDirName             = "pp"
+)
+
 var (
 	nodeGenesisCommand = &cli.Command{
 		Name:    "generate-node-genesis",
 		Aliases: []string{"node-genesis", "genesis"},
 		Usage:   "Create the genesis configuration file as expected by the node",
-		Action:  nodeGenesis,
+		Action:  createNodeGenesis,
 		Flags: []cli.Flag{
 			l1Flag,
 			outputFlag,
-			&cli.StringFlag{
-				Name:    rollupManagerAliasFlagName,
-				Aliases: []string{"rm"},
-				Usage: fmt.Sprintf(
-					"Name of the rollup manager to which the rollup belongs. Needs to match an already imported rollup manager (can be done by running the %s command)",
-					importRollupManagerCommandName,
-				),
-				Required: true,
-			},
-			&cli.StringFlag{
-				Name:    rollupAliasFlagName,
-				Aliases: []string{"r"},
-				Usage: fmt.Sprintf(
-					"Name of the rollup. Needs to match an already imported rollup (can be done by running the %s command)",
-					importRollupCommandName,
-				),
-				Required: true,
-			},
+			rollupManagerAliasFlag,
+			rollupAliasFlag,
 		},
 	}
 )
 
-func nodeGenesis(cliCtx *cli.Context) error {
+type l1Config struct {
+	ChainId       uint64         `json:"chainId"`
+	Rollup        common.Address `json:"polygonZkEVMAddress"`
+	RollupManager common.Address `json:"polygonRollupManagerAddress"`
+	POL           common.Address `json:"polTokenAddress"`
+	GER           common.Address `json:"polygonZkEVMGlobalExitRootAddress"`
+}
+
+type nodeGenesis struct {
+	L1Config                                l1Config    `json:"l1Config"`
+	RollupCreationBlockNumberUsedByRollup   uint64      `json:"genesisBlockNumber"`
+	RollupCreationBlockNumberUsedByValidium uint64      `json:"rollupCreationBlockNumber"`
```

Comment:
This naming is strange, both are the same concept but the json serialization is completely different?

Reply:
Indeed it is strange. It is because, for the rollups, the rollup creation was named the `genesisBlockNumber` (and it was renamed for the validiums to `rollupCreationBlockNumber`).

I have left the `rollupCreationBlockNumber`, because it seems more consistent (3387e2e6a40c76440b3791893210f8d55488713b). 

As a side note, I'm unsure if the genesis command is still in use, because we have moved to the CDK Erigon.

Reply:
Feedback from the V team:

> I'm not entirely sure about how the components use the genesisBlockNumber or rollupCreationBlockNumber fields, but I'd imagine that they just ignore the field that is not needed. In that case I think it's fine just to leave both!

So we are not going to remove them, because as I've mentioned, the `genesisBlockNumber` is used by the rollup chains, and the `rollupCreationBlockNumber` by the validium chains and the tool is in use for both chains still.